radioactivedecay/inventory.py

```diff
@@ -293,58 +620,67 @@ def __len__(self) -> int:
 
         return len(self.contents)
 
-    def add(self, add_contents: Dict[Union[str, Radionuclide], float]) -> None:
+    def add(
+        self,
+        add_contents: Dict[Union[str, Radionuclide], float],
+        input_type: str = "activity",
+    ) -> None:
         """
-        Adds a dictionary of radionuclides and associated activities to the inventory.
+        Adds a dictionary of radionuclides and associated numbers to the inventory.
 
         Parameters
         ----------
         add_contents : dict
             Dictionary containing radionuclide strings or Radionuclide objects as keys and
             activities as values which are added to the Inventory.
+        input_type : str, optional
+            Type of input, default is activities.
 
         Examples
         --------
-        >>> inv = rd.Inventory({'H-3': 1.0})
-        >>> inv.add({'C-14': 2.0})
-        >>> inv.contents
+        >>> inv = rd.Inventory({'H-3': 1.0}, "numbers")
+        >>> inv.add({'C-14': 2.0}, "numbers")
+        >>> inv.contents()
         {'C-14': 2.0, 'H-3': 1.0}
 
         """
 
         parsed_add_contents: Dict[str, float] = _check_dictionary(
-            add_contents, self.data.radionuclides, self.data.dataset
+            add_contents, self.data.radionuclides, input_type, self.data
         )
         new_contents = _add_dictionaries(self.contents, parsed_add_contents)
-        self._change(new_contents, False, self.data)
+        self._change(new_contents, "numbers", False, self.data)
 
-    def subtract(self, sub_contents: Dict[Union[str, Radionuclide], float]) -> None:
+    def subtract(
+        self,
+        sub_contents: Dict[Union[str, Radionuclide], float],
+        input_type: str = "activities",
+    ) -> None:
         """
-        Subtracts a dictionary of radionuclides and associated activities from this inventory.
+        Subtracts a dictionary of radionuclides and associated numbers from this inventory.
 
         Parameters
         ----------
         sub_contents : dict
             Dictionary containing radionuclide strings or Radionuclide objects as keys and
-            activities as values which are subtracted from the Inventory.
+            numbers as values which are subtracted from the Inventory.
```

Comment:
This isn't just numbers. It's what the user specifies via input_type. We need to add as well:
```
input_type : str, optional
        Type of input (masses, numbers, activities; activities is default)
```